Weird error when implementing Send with a nonexisting type param

```rust
struct Foo<T>(T, *const ());

unsafe impl Send for Foo<T> {}
```
results in
```
error[E0412]: cannot find type `T` in this scope
 --> src/lib.rs:3:26
  |
3 | unsafe impl Send for Foo<T> {}
  |                          ^ not found in this scope

error[E0277]: `*const ()` cannot be sent between threads safely
  --> src/lib.rs:3:13
   |
3  |   unsafe impl Send for Foo<T> {}
   |               ^^^^ `*const ()` cannot be sent between threads safely
   |
   = help: within `Foo<[type error]>`, the trait `std::marker::Send` is not implemented for `*const ()`
   = note: required because it appears within the type `Foo<[type error]>`
```
The second error is incorrect and quite confusing.

--- a/compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs
+++ b/compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs
@@ -1906,6 +1906,11 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             ObligationCauseCode::BuiltinDerivedObligation(ref data) => {
                 let parent_trait_ref = self.resolve_vars_if_possible(&data.parent_trait_ref);
                 let ty = parent_trait_ref.skip_binder().self_ty();
+                if parent_trait_ref.references_error() {
+                    err.cancel();
+                    return;
+                }
+
                 err.note(&format!("required because it appears within the type `{}`", ty));
                 obligated_types.push(ty);
 

--- a/src/test/ui/traits/issue-75627.rs
+++ b/src/test/ui/traits/issue-75627.rs
@@ -1,0 +1,6 @@
+struct Foo<T>(T, *const ());
+
+unsafe impl Send for Foo<T> {}
+//~^ ERROR cannot find type
+
+fn main() {}

--- a/src/test/ui/traits/issue-75627.stderr
+++ b/src/test/ui/traits/issue-75627.stderr
@@ -1,0 +1,9 @@
+error[E0412]: cannot find type `T` in this scope
+  --> $DIR/issue-75627.rs:3:26
+   |
+LL | unsafe impl Send for Foo<T> {}
+   |                          ^ not found in this scope
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0412`.

